lib/valueflow.cpp

```diff
@@ -5116,7 +5121,8 @@ static void valueFlowConditionExpressions(TokenList *tokenlist, SymbolDatabase*
             {
                 for (const Token* condTok2 : getConditions(condTok, "&&")) {
                     if (is1) {
-                        SameExpressionAnalyzer a1(condTok2, makeConditionValue(1, condTok2, true), tokenlist);
+                        const bool isBool = astIsBool(condTok2->astOperand1()) && astIsBool(condTok2->astOperand2());
```

Comment:
This won't work with non-bool variables. This should be `astIsBool(condTok2) || Token::Match(condTok2, "%comp%|%oror%|&&")`.